src/coreclr/src/vm/dllimportcallback.cpp

```diff
@@ -961,9 +961,16 @@ void UMEntryThunk::Terminate()
     CONTRACTL
     {
         NOTHROW;
+        MODE_ANY;
     }
     CONTRACTL_END;
 
+    if (GetObjectHandle())
+    {
+        DestroyLongWeakHandle(GetObjectHandle());
+        this->m_pObjectHandle = 0;
```

Comment:
Nit: The rest of this file does not use `this->` to access fields, so this place should not either for consistent style.